solar/solar/core/handlers/shell.py

```diff
@@ -21,4 +22,19 @@
 class Shell(TempFileHandler):
     def action(self, resource, action_name):
         action_file = self._compile_action_file(resource, action_name)
-        fabric_api.local('bash {}'.format(action_file))
+        log.debug('action_file: %s', action_file)
+
+        self.transport_sync.copy(resource, action_file, '/tmp/action.sh')
```

Comment:
Please do not call it action.sh
resource_name_action.sh will help with debug. 
We also need to fix it in puppet